src/components/ProfileTable.vue

```diff
@@ -1,23 +1,143 @@
+              @click="colorPickerVisible = !colorPickerVisible"
+              v-model="activeProfile.profile.custom!.color"
+            />
+            <InputText v-if="activeProfile.applicationName === 'AWS Account'"
+              v-model="activeProfile.profile.custom!.color" class="p-inputtext-sm"
+              style="width: 100px; margin-left: 10px"
+            />
+            <!--- Colorpicker popup doesn't work, create our own --->
+            <PDialog
+              v-if="$ext.platform === 'firefox' || $ext.platform === 'safari'"
+              v-model:visible="colorPickerVisible"
+              :style="{ width: '50vw' }"
+            >
+              <ColorPicker v-if="colorPickerVisible" :inline="true" v-model="activeProfile.profile.custom!.color" />
+            </PDialog>
+          </div>
+          <div
+            v-if="activeProfile.applicationName === 'AWS Account' && (!permissions.console || !permissions.signin)"
+            style="text-align: center; padding-top: 20px;"
+          >
+            <PrimeButton size="small" icon="pi pi-lock" class="p-button-success" label="Request Permissions"
+              style="margin-top: 5px" @click="requestPermissions()" />
+            <p style="margin-top: 10px;">
+              In order to customize the AWS console and assume IAM roles, this extension requires additional permissions.
+            </p>
+          </div>
+          <div v-else-if="activeProfile.applicationName === 'AWS Account'">
+            <div style="margin-bottom: 10px">
+              <small id="account-label-help">AWS Account ID</small>
+              <InputText
+                v-model="activeProfile.searchMetadata!.AccountId"
+                class="p-inputtext-sm"
+                aria-describedby="account-label-help"
+                style="width: 400px"
+                :disabled="true"
+              />
+            </div>
+            <div style="margin-bottom: 10px">
+              <small id="account-name-label-help">AWS Account Name</small>
+              <InputText
+                v-model="activeProfile.searchMetadata!.AccountName"
+                class="p-inputtext-sm"
+                aria-describedby="account-name-label-help"
+                style="width: 400px"
+                :disabled="true"
+              />
+            </div>
+            <div
+              style="margin-bottom: 10px"
+            >
+              <small id="profile-label-help">AWS Console Preview</small>
+              <InputText
+                id="consolePreview"
+                v-model="consolePreview"
+                class="p-inputtext-sm"
+                aria-describedby="profile-label-help"
+                style="width: 400px"
+                :disabled="true"
+                :style="consoleStyle"
+              />
+            </div>
+          </div>
+          <div>
+            <h3 v-if="activeProfile.profile.custom!.iamRoles.length > 0">IAM Roles</h3>
```

Comment:
Hi @WTFender , I'm getting some issues on this line, I haven't had enough time to debug but the error is:
```
TypeError: Cannot read properties of undefined (reading 'length')
```

I believe the issue here is that iamRole can be null (and you also use the `!.`) and this case I believe you will need to use `?.length` to avoid the error. `undefined > 0` will also be false which should work as expected.

Reply:
@pmarques You may be right, but I don't see this error in my IDE or when building with `npm run build:chrome`.

I don't think there is much harm in adding the `?`, so just pushed it to main.